### DISCUSSION TITLE
ENV returns only a chunk of environment variables

### DISCUSSION THREAD

**Discussion by gwuah** [Homebrew/discussions#3335] [Everyday usage] [answered]
So i'm working on a formula and im unable to read an enviroment variable that I set.
I tried logging `ENV.keys` in my formula.rb file and this is what was returned.
None of my other  env variables appeared in this list.
Any help?

OS: MacOS Monterey, 12.4
Apple M1 chip

```
["HOMEBREW_BREW_DEFAULT_GIT_REMOTE", "HOMEBREW_DEFAULT_CACHE", "HOMEBREW_PROCESSOR", "HOMEBREW_DEV_CMD_RUN", "HOMEBREW_USER_AGENT", "HOMEBREW_AUTO_UPDATE_CHECKED", "HOMEBREW_COMMAND_DEPTH", "HOMEBREW_LIBRARY", "HOMEBREW_TEMP", "TERM", "SHELL", "HOMEBREW_OS_VERSION", "HOMEBREW_REPOSITORY", "HOMEBREW_RUBY_PATH", "HOMEBREW_GIT_CONFIG_FILE", "LC_ALL", "HOMEBREW_RUBY_WARNINGS", "HOMEBREW_PATH", "USER", "HOMEBREW_CORE_GIT_REMOTE", "SSH_AUTH_SOCK", "HOMEBREW_ANALYTICS_USER_UUID", "HOMEBREW_MINIMUM_GIT_VERSION", "HOMEBREW_CACHE", "HOMEBREW_DEFAULT_LOGS", "HOMEBREW_DEVELOPER_MODE", "HOMEBREW_USER_AGENT_CURL", "PATH", "HOMEBREW_BREWED_CURL_PATH", "PWD", "HOMEBREW_BREW_FILE", "HOMEBREW_MACOS_VERSION_NUMERIC", "HOMEBREW_MACOS_VERSION", "HOMEBREW_REQUIRED_RUBY_VERSION", "HOMEBREW_DEFAULT_TEMP", "HOMEBREW_PRODUCT", "HOMEBREW_AUTO_UPDATING", "HOMEBREW_GITHUB_PACKAGES_AUTH", "SHLVL", "HOME", "HOMEBREW_BREW_GIT_REMOTE", "HOMEBREW_PREFIX", "HOMEBREW_BOTTLE_DEFAULT_DOMAIN", "LOGNAME", "HOMEBREW_GIT", "HOMEBREW_LOGS", "HOMEBREW_VERSION", "HOMEBREW_CELLAR", "HOMEBREW_MACOS_SYSTEM_RUBY_NEW_ENOUGH", "HOMEBREW_CURL", "HOMEBREW_ANALYTICS_IDS", "HOMEBREW_SYSTEM", "HOMEBREW_BIN", "__CF_USER_TEXT_ENCODING", "TMPDIR"]
```

**Comment by SMillerDev** [comment]
Did you set it in the formula? Otherwise it's filtered out.

**Reply by gwuah** [reply]
one of the reasons i'm setting it as an environment variable is because i don't want it to be committed to git. (private keys)
any reason why env variables are being filtered out?

also goes without saying, it was working fine before, and after an upgrade, it stopped working.

**Reply by SMillerDev** [reply]
They're filtered out because often the environment variables people set cause their builds to fail.

**Reply by gwuah** [reply]
found the answer here.
thanks for your help https://github.com/orgs/Homebrew/discussions/3335#discussioncomment-2891286

**Comment by gwuah** [comment, marked as answer]
Okay so i just figured the solution (thanks to a comment by @SMillerDev )
Turns out ENV variables that are not prefixed with `HOMEBREW_` are filtered out.
